src/components/CommentCard.jsx

```diff
@@ -1,11 +1,55 @@
-export const CommentCard = ({author, created_at, body, votes}) => {
+import { useContext, useState, useEffect } from "react"
+import { toast } from "react-toastify";
+import "react-toastify/dist/ReactToastify.css";
+import { UserContext } from "../contexts/User";
+import { deleteComment } from "../utils/api";
+
+
+
+
+export const CommentCard = ({author, created_at, body, votes, comment_id, setComments}) => {
+    const userForCommentDelete = useContext(UserContext);
+    const [commentIdToDelete, setCommentIdToDelete] = useState(comment_id);
+    const [isClicked, setIsClicked] = useState(false);
+  
+
+    useEffect(() => {
+        if(isClicked) {
+        deleteComment(commentIdToDelete).then(() => {
+            optimisticDeleteComment(commentIdToDelete) 
+        })
+        .catch((err) => {
+            handleError()   
+        })
+        }
+    }, [isClicked])
```

Comment:
Does this need a use effect?

Currently, you're clicking a button, to change to state, the state change is then triggering a use effect.

Could you think of a refactor where you can click the button, and it invokes the function? - You already have the function that needs to be invoked - there no real need for the useEffect.